TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -905,6 +909,149 @@ private void WritePeptideResults()
             }
         }
 
+        private void WriteVariantResults()
+        {
+            Status("Writing variant peptide results...", Parameters.SearchTaskId);
+            string variantPsmFile = Path.Combine(Parameters.OutputFolder, "VariantPSMs.psmtsv");
+            string variantPeptideFile = Path.Combine(Parameters.OutputFolder, "VariantPeptides.psmtsv");            
+            List<PeptideSpectralMatch> FDRPsms = Parameters.AllPsms
+                .Where(p => p.FdrInfo.QValue <= CommonParameters.QValueOutputFilter
+                && p.FdrInfo.QValueNotch <= CommonParameters.QValueOutputFilter && p.BaseSequence != null).ToList();
+            List<PeptideSpectralMatch> possibleVariantPsms = new List<PeptideSpectralMatch>();
+            foreach (var peptide in FDRPsms)
+            {
+                var variants = peptide.BestMatchingPeptides.SelectMany(p => p.Peptide.Protein.AppliedSequenceVariations.Select(v => PsmTsvWriter.IntersectsWithVariation(p.Peptide, v, true)).ToList()).ToList();
```

Comment:
`bool psmContainsVariant = psm.BestMatchingPeptides.Any(p => p.Peptide.Protein.AppliedSequenceVariations.Any(v => PsmTsvWriter.IntersectsWithVariation(p.Peptide, v, true)));` should be a bit more efficient. 

`IntersectsWithVariation` looks like it belongs more in `PeptideWithSetModifications` in mzLib, rather than the `PsmTsvWriter`. it doesn't have anything to do with PSM writing. it looks like the `Includes` method in `PsmTsvWriter` also belongs in `PeptideWithSetModifications`. maybe also the `SequenceVariantString` method.

Reply:
Good point. I think we should do this.

I started it here: https://github.com/smith-chem-wisc/mzLib/pull/508

Moving it makes it much easier to test, too, and some issues are becoming apparent:
1. stop losses aren't being included in the protein XML; we'll need to look into the SnpEff method for making this to see if we can include them.
2. what is a variant peptide identification? It's a subtle point, but I think it needs to cross a part of the sequence that contains a sequence change. Otherwise, it's not really evidence for that sequence variation. An example of this would be an MNP where the sequence changes from "AA" to "AR". If you only cross the first A, it's not a sequence variant peptide identification because it doesn't cross something that has a sequence change. If we agree that's the case, I think we could probably remove the boolean check for "checkForSequenceChange" and just make that our assumption. @rmmiller22, what do you think about this?

Reply:
I agree with the point that to be a variant containing peptide the sequence must fully cross the variant.  I feel like being more stringent with what we classify as a variant peptide is a good thing, and making the assumption that the sequence changes is fine.